src/components/Countdown/Countdown.tsx

```diff
@@ -0,0 +1,151 @@
+import { ComponentProps, forwardRef, useCallback, useEffect, useState } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface CountdownProps extends ComponentProps<'div'> {
+  /**
+   * The date the countdown should end
+   */
+  endDate: string;
```

Comment:
I think it's probably best to take a `Date` type here since otherwise you'd have to worry about inconsistencies in parsing based on different formats